.github/instructions/buffer-indicators.instructions.md

```diff
@@ -0,0 +1,274 @@
+    
+    foreach (TQuote quote in quotes)
+    {
+        bufferList.Add(quote);
+    }
+    
+    return bufferList;
+}
+```
+
+## Testing requirements
+
+### Test coverage expectations
+
+Buffer indicator tests must cover:
+
+1. **Incremental processing** - Verify correct calculation as quotes are added
+2. **Buffer capacity** - Test behavior at and beyond buffer limits
+3. **State management** - Verify Clear() method resets state properly
+4. **Memory efficiency** - Confirm no memory leaks or excessive allocations
+5. **Performance benchmarks** - Must not exceed expected performance thresholds
+
+### Test structure pattern
+
+```csharp
+[TestClass]
+public class {IndicatorName}BufferListTests : TestBase
+{
+    [TestMethod]
+    public void Add()
+    {
+        // Test incremental addition
+        {IndicatorName}BufferList bufferList = new(lookbackPeriods);
+        
+        foreach (Quote quote in quotes.Take(50))
+        {
+            {IndicatorName}Result result = bufferList.Add(quote);
+            
+            // Verify incremental results
+            if (shouldHaveValue)
+            {
+                result.{Property}.Should().BeApproximately(expectedValue, precision);
+            }
+        }
+    }
+
+    [TestMethod]
+    public void AddWithInsufficientData()
+    {
+        {IndicatorName}BufferList bufferList = new(lookbackPeriods);
+        
+        // Add fewer quotes than required
+        foreach (Quote quote in quotes.Take(lookbackPeriods - 1))
+        {
+            {IndicatorName}Result result = bufferList.Add(quote);
+            result.{Property}.Should().BeNull();
+        }
+    }
+
+    [TestMethod]
+    public void Clear()
+    {
+        {IndicatorName}BufferList bufferList = quotes.To{IndicatorName}BufferList();
+        bufferList.Clear();
+        
+        // Verify state is reset
+        {IndicatorName}Result result = bufferList.Add(quotes.First());
+        result.{Property}.Should().BeNull();
+    }
+
+    [TestMethod]
+    public void FullyPopulated()
+    {
+        // Compare buffer results with series results for perfect match
+        IEnumerable<{IndicatorName}Result> bufferResults = quotes.To{IndicatorName}BufferList();
+        IEnumerable<{IndicatorName}Result> seriesResults = quotes.To{IndicatorName}();
+        
+        bufferResults.Should().BeEquivalentTo(seriesResults);
+    }
+}
+```
+
+### Performance benchmarking
+
+Buffer indicators must include performance tests in the `tests/performance` project that verify efficiency:
+
+```csharp
+// In tests/performance project
+[MethodImpl(MethodImplOptions.NoInlining)]
+public void BufferIndicator{IndicatorName}()
+{
+    {IndicatorName}BufferList bufferList = new(lookbackPeriods);
+    
+    foreach (Quote quote in quotes)
+    {
+        _ = bufferList.Add(quote);
+    }
+}
+```
+
+**Performance expectations**:
+
+- Buffer processing should be significantly faster than series processing
```

Comment:
This not actually true for normal quantities of input values, run some same-indicator samples to see expectations.